src/NUnitFramework/framework/Internal/Commands/DisposeFixtureCommand.cs

```diff
@@ -41,15 +41,17 @@ public class DisposeFixtureCommand : AfterTestCommand
         public DisposeFixtureCommand(TestCommand innerCommand)
             : base(innerCommand)
         {
-            Guard.OperationValid(Test is IDisposableFixture, "DisposeFixtureCommand does not apply to a " + Test.GetType().Name);
```

Comment:
Another point where the intelligence is moved __into__ the command rather than resting in the command chain generation logic.